src/modules/fancyzones/lib/ZoneWindow.cpp

```diff
@@ -9,6 +9,63 @@
 #include "RegistryHelpers.h"
 
 #include <ShellScalingApi.h>
+#include <mutex>
+
+namespace {
+  enum class TTmpFileType {
+    EActiveZoneSetFile = 0,
+    EAppliedZoneSetFile,
+    ECustomZoneSetsFile
+  };
+
+  std::wstring GenerateTmpFileName(TTmpFileType type)
+  {
+    static std::wstring activeZoneSetTmpFileName;
+    static std::wstring appliedZoneSetTmpFileName;
+    static std::wstring customZoneSetsTmpFileName;
+    static std::once_flag f1, f2, f3;
+
+    std::call_once(f1, []() {
```

Comment:
We should put string initialization next to their declaration. The `f1` etc. should be named more meaningful.